### PR TITLE
Fix `Delegate` dispatch.

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
/cc @janvorli @cshung

### CODE REVIEW

File: src/coreclr/vm/comdelegate.cpp [hunk 1]
```diff
@@ -1901,16 +1905,21 @@ Stub* COMDelegate::GetInvokeMethodStub(EEImplMethodDesc* pMD)
...
             pCode->EmitLDARG(paramCount);
 
-        // recursively call the delegate itself
+#ifdef FEATURE_INTERPRETER
+        // Call the underlying method pointer.
+        pCode->EmitLoadThis();
+        pCode->EmitLDFLD(FIELD__DELEGATE__METHOD_PTR);
+
+        mdToken sigTok = pCode->GetSigToken(pSig, cbSig);
+        pCode->EmitCALLI(sigTok, sig.NumFixedArgs(), fReturnVal);
```
Comment (huoyaoyuan) [inline src/coreclr/vm/comdelegate.cpp:1908]: Is there any concern about GC safety of collectible methods? (dotnet/runtime#105082)

I'm not familiar about how interpreter interacts with GC.

Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/comdelegate.cpp:1908]: It really shouldn't impact the interpreter scenario, but it costs us nothing to make it correct. This is throw away code right now, but I will make it correct. Thanks for pointing this out.

Comment (huoyaoyuan) [reply src/coreclr/vm/comdelegate.cpp:1908]: I originally used the same mechanism in dotnet/runtime#104731 . You may keep the mentioned points in mind when bringing the interpreter for production quality. /cc @jkotas

Comment (jkotas) [reply src/coreclr/vm/comdelegate.cpp:1908]: This is throw-away code. We do not plan to bring this implementation of the interpreter to quality. The change looks good as a temporary measure. It has multiple corner-case problems and that's fine. 

The only 100% correct fix would be to treat `Invoke` as an intrinsic in the interpreter (ie keep this as is and change the code in interpreter.cpp instead).

**Review by jkotas — APPROVED** [review]

**Pull request merged**